src/services/codefixes/convertFunctionToEs6Class.ts

```diff
@@ -71,18 +56,31 @@ namespace ts.codefix {
                             isObjectLiteralExpression(firstDeclaration.parent.right)
                         ) {
                             const prototypes = firstDeclaration.parent.right;
-                            const memberElement = createClassElement(prototypes.symbol, /** modifiers */ undefined);
-                            if (memberElement) {
-                                memberElements.push(...memberElement);
-                            }
+                            createClassElement(prototypes.symbol, /** modifiers */ undefined, memberElements);
                         }
                     }
                     else {
-                        const memberElement = createClassElement(member, [factory.createToken(SyntaxKind.StaticKeyword)]);
-                        if (memberElement) {
-                            memberElements.push(...memberElement);
+                        createClassElement(member, [factory.createToken(SyntaxKind.StaticKeyword)], memberElements);
+                    }
+                });
+            }
+
+            // all instance members are stored in the "member" array of symbol (done last so instance members pulled from prototype assignments have priority)
+            if (symbol.members) {
+                symbol.members.forEach((member, key) => {
+                    if (key === "constructor" && member.valueDeclaration) {
+                        const prototypeAssignment = symbol.exports?.get("prototype" as __String)?.declarations?.[0]?.parent;
+                        if (prototypeAssignment && isBinaryExpression(prototypeAssignment) && isObjectLiteralExpression(prototypeAssignment.right) && some(prototypeAssignment.right.properties, isConstructorAssignment)) {
+                            // fn.prototype = { constructor: fn }
+                            // Already deleted in `createClassElement` in first pass

```

Comment:
Likewise, convertToES6Class had an issue where a js prototype property merge sometimes happens and sometimes does not (and it breaks when the merge _does_ occur in `main`). Since I didn't like crashing fourslash tests, in lieu of fixing the underlying issue (making the merge always occur regardless of what symbol is requested as an entrypoint), I've hardened the implementation to work when a symbol appears as both a `.prototype` member _and_ has been merged into the actual `symbol.members` set (and, coincedentally, moved static members to before instance members in the generated class, which I much prefer). Just like with the only-sometimes-merged js baseline symbol, I think fixing that (existing) bug should be a separate issue.